### PR TITLE
Move PlanWizard to its own page

### PR DESCRIPTION
**Pull request by gildub**
Resolves https://github.com/konveyor/virt-ui/issues/54

### CODE REVIEW

File: src/app/Plans/PlansPage.tsx [hunk 1]
```diff
@@ -75,13 +73,11 @@ const PlansPage: React.FunctionComponent = () => {
...
                   >
                     <div className={`${spacing.mtMd}`}>
-                      <Button
-                        isDisabled={addPlanDisabledObj.isTooltipEnabled}
-                        onClick={toggleWizard}
-                        variant="primary"
-                      >
-                        Create migration plan
-                      </Button>
+                      <Link to="/planwizard">
```
Comment (mturley) [inline src/app/Plans/PlansPage.tsx:76]: Let's see if we can keep the URL paths roughly based on the breadcrumb bar, so this should be a route under `/plans`. Maybe `/plans/create`?

Comment (mturley) [reply src/app/Plans/PlansPage.tsx:76]: Also does this link properly disable with the button is disabled? Not sure how that works when it's wrapped around it like this. If not we may need to either only render the <Link> around it when the button would be enabled, or handle the link behavior as an onClick instead.

File: src/app/Plans/components/PlansTable.tsx [hunk 2]
```diff
@@ -172,9 +169,9 @@ const PlansTable: React.FunctionComponent<IPlansTableProps> = ({
     <>
       <Level>
         <LevelItem>
-          <Button variant="secondary" onClick={toggleWizard} isDisabled={false}>
-            Create
-          </Button>
+          <Link to="/planwizard">
```
Comment (mturley) [inline src/app/Plans/components/PlansTable.tsx:172]: See above re: maybe using `/plans/create` instead of `/planwizard`. Thoughts on that @ibolton336 ?

File: src/app/Plans/components/Wizard/PlanWizard.tsx [hunk 3]
```diff
@@ -121,16 +114,23 @@ const PlanWizard: React.FunctionComponent<IPlanWizardProps> = ({
   };
 
   return (
-    <Modal isOpen width="95%" showClose={false} hasNoBodyWrapper>
+    <PageSection variant="light">
+      <Breadcrumb className={`${spacing.mbLg} ${spacing.prLg}`}>
+        <BreadcrumbItem>
+          <Link to={`/plans`}>Migration plans</Link>
+        </BreadcrumbItem>
+        <BreadcrumbItem>Plan Wizard</BreadcrumbItem>
```
Comment (mturley) [inline src/app/Plans/components/Wizard/PlanWizard.tsx:122]: The word "wizard" shouldn't really be shown in the UI as the name of the page, right @vconzola ? I imagine this breadcrumb bar should read "Migration plans > Create" ?

File: src/app/Plans/components/Wizard/PlanWizard.tsx [hunk 4]
```diff
@@ -121,16 +114,23 @@ const PlanWizard: React.FunctionComponent<IPlanWizardProps> = ({
...
+      </Breadcrumb>
       <Wizard
         title="Create a Migration Plan"
         steps={steps}
-        onClose={onClose}
         onNext={onMove}
         onBack={onMove}
         onSubmit={(event) => event.preventDefault()}
+        hideClose={true}
+        cancelButtonText=""
```
Comment (mturley) [inline src/app/Plans/components/Wizard/PlanWizard.tsx:131]: This hides the Cancel button. @vconzola do we still want the cancel button in place? It could just bring you back to the plans page.

File: src/app/Plans/components/Wizard/PlanWizard.tsx [hunk 5]
```diff
@@ -121,16 +114,23 @@ const PlanWizard: React.FunctionComponent<IPlanWizardProps> = ({
...
-    <Modal isOpen width="95%" showClose={false} hasNoBodyWrapper>
+    <PageSection variant="light">
+      <Breadcrumb className={`${spacing.mbLg} ${spacing.prLg}`}>
+        <BreadcrumbItem>
+          <Link to={`/plans`}>Migration plans</Link>
+        </BreadcrumbItem>
+        <BreadcrumbItem>Plan Wizard</BreadcrumbItem>
+      </Breadcrumb>
       <Wizard
         title="Create a Migration Plan"
```
Comment (mturley) [inline src/app/Plans/components/Wizard/PlanWizard.tsx:125]: Let's remove the `title` prop and instead add a title to the page section above, by the breadcrumb bar like we have on our other pages. The black title bar looks a little awkward embedded on the page.

File: src/app/routes.tsx [hunk 6]
```diff
@@ -62,6 +63,12 @@ export const routes: AppRouteConfig[] = [
     path: '/plans',
     title: `${APP_TITLE} | Migration Plans`,
   },
+  {
+    component: PlanWizard,
+    exact: true,
+    path: '/planwizard',
```
Comment (mturley) [inline src/app/routes.tsx:69]: See above re: the route path

**Review by mturley — CHANGES_REQUESTED** [review]
Looking good! A couple of little things.

File: src/app/routes.tsx (hunk 6 above)
Comment (mturley) [reply src/app/routes.tsx:69]: One thing to be aware of is if we make this `/plans/create` I believe it needs to be higher in the list than `/plans/:planId` in order to prevent matching that route instead, so let's stay aware of that when this change gets rebased on/under #78.

**Comment by mturley** [comment]
@vconzola what do you think about putting the wizard in a Card? if we move the title up into the top page section above the breadcrumb bar, we could give the lower section the grey background like our other pages have, with the wizard in a card on top of that, instead of having the sections both have white backgrounds and blend together.

File: src/app/Plans/components/Wizard/PlanWizard.tsx (hunk 3 above)
Comment (gildub) [reply src/app/Plans/components/Wizard/PlanWizard.tsx:122]: Looks better, I've changed and aligned it to go along with above comments.

@vconzola to confirm it or use something else.

File: src/app/Plans/components/Wizard/PlanWizard.tsx (hunk 4 above)
Comment (gildub) [reply src/app/Plans/components/Wizard/PlanWizard.tsx]: I've restored it to help @vconzola review it.

File: src/app/Plans/PlansPage.tsx (hunk 1 above)
Comment (gildub) [reply src/app/Plans/PlansPage.tsx]: That's a good point. I'll done a quick unary fix before adding a wrapper for long run.

File: src/app/routes.tsx (hunk 6 above)
Comment (gildub) [reply src/app/routes.tsx]: I've added an TODO comment too to help.

File: src/app/Plans/PlansPage.tsx [hunk 7]
```diff
@@ -75,13 +73,11 @@ const PlansPage: React.FunctionComponent = () => {
...
                     <div className={`${spacing.mtMd}`}>
-                      <Button
-                        isDisabled={addPlanDisabledObj.isTooltipEnabled}
-                        onClick={toggleWizard}
-                        variant="primary"
-                      >
-                        Create migration plan
-                      </Button>
+                      <Link to="/plans/create">
+                        <Button isDisabled={addPlanDisabledObj.isTooltipEnabled} variant="primary">
```
Comment (mturley) [inline src/app/Plans/PlansPage.tsx:79]: @gildub I tested rendering this button with `isDisabled={true}` and the Link still takes you to the wizard page. Also, `<Link>` renders an `<a>` and having a `<button>` inside an `<a>` is not semantically allowed in HTML (even though it renders, we're probably getting a warning about it).

Instead we should probably just use the button and navigate in its `onClick` handler. You can use the `useHistory` hook from `react-router-dom` and do `useHistory().push('/plans/create')`. https://reactrouter.com/web/api/Hooks/usehistory

Comment (gildub) [reply src/app/Plans/PlansPage.tsx:78]: Oh I see re having `<button>` embedded in a `<a>`. I was actually in unknown territory for me when I did that this morning. 
I guess I pick up a wrong approach. Good to know, thanks.

Let me use the `useHistory`.

File: src/app/Plans/PlansPage.tsx [hunk 8]
```diff
@@ -10,7 +10,7 @@ import {
   Button,
   Alert,
 } from '@patternfly/react-core';
-import { Link } from 'react-router-dom';
+import { Link, useHistory } from 'react-router-dom';
```
Comment (mturley) [inline src/app/Plans/PlansPage.tsx:13]: This Link import is unused now, let's remove it

File: src/app/Plans/PlansPage.tsx [hunk 9]
```diff
@@ -28,9 +28,11 @@ const plans = MOCK_PLANS;
 const migrations = MOCK_MIGRATIONS;
 
 const PlansPage: React.FunctionComponent = () => {
+  const history = useHistory();
   const providersQuery = useProvidersQuery();
 
-  const vmwareProviders = providersQuery.data?.vsphere || [];
+  // const vmwareProviders = providersQuery.data?.vsphere || [];
```
Comment (mturley) [inline src/app/Plans/PlansPage.tsx:34]: Can you revert this change and go back to using the real data for `vmwareProviders`?

File: src/app/queries/mocks/plans.mock.ts [hunk 10]
```diff
@@ -231,7 +231,7 @@ const plan4: IPlan = {
   },
 };
 
-export const MOCK_PLANS: IPlan[] = [plan1, plan2, plan3, plan4];
+export const MOCK_PLANS: IPlan[] = []; //[plan1, plan2, plan3, plan4];
```
Comment (mturley) [inline src/app/queries/mocks/plans.mock.ts:234]: Same here, I assume this was to test the empty state, can we revert it?

**Review by mturley — CHANGES_REQUESTED** [review]
Last thing!

File: src/app/Plans/PlansPage.tsx (hunk 8 above)
Comment (gildub) [reply src/app/Plans/PlansPage.tsx:13]: Of course!

File: src/app/Plans/PlansPage.tsx (hunk 9 above)
Comment (gildub) [reply src/app/Plans/PlansPage.tsx:34]: Darn me!

File: src/app/queries/mocks/plans.mock.ts (hunk 10 above)
Comment (gildub) [reply src/app/queries/mocks/plans.mock.ts:234]: Same!

**Review by mturley — APPROVED** [review]
Excellent!
